### DISCUSSION TITLE
Low framerate on dynamic content despite ideal network conditions and settings

### DISCUSSION THREAD

**Discussion by JPenuchot** [LizardByte/community#18] [Sunshine] [answered]
### Describe the Bug

Moonlight reports incoming framerates from network that are lower than expected (sometimes under 30FPS) on anything that isn't a static image. This happens with a bitrate cap of 150mbps (max allowed by Moonlight), network speeds above 500mbps, and latency under 2ms, and it happens regardless of the encoder settings (high performance, llhq, cbr, etc...).

Reported in-game framerates are well above 60FPS, but Sunshine will simply drop half of them.

### Expected Behavior

I would expect a steady 60FPS, considering the network conditions are optimal (or close to it).

### Sunshine Host Operating System and Version

Arch

### Architecture

64bit

### Sunshine Version

0.13.0

### GPU Type

Nvidia

### GPU Model

GTX 970

### GPU Driver/Mesa Version

510.54

### Capture Method (Linux Only)

X11

**Comment by KuleRucket** [comment]
Do you have NvFBC working?

**Reply by JPenuchot** [reply]
I'm not able to get Sunshine to run with patched utils. Here's the output:

```
Mar 21 10:47:47 jpenuchot-nzxt sunshine[1400]: [2022:03:21:10:47:47]: Info: CLIENT CONNECTED
Mar 21 10:47:47 jpenuchot-nzxt sunshine[1400]: [2022:03:21:10:47:47]: Info: Found [1] outputs
Mar 21 10:47:47 jpenuchot-nzxt sunshine[1400]: [2022:03:21:10:47:47]: Info: Virtual Desktop: 1920x1080
Mar 21 10:47:47 jpenuchot-nzxt sunshine[1400]: [2022:03:21:10:47:47]: Info: XrandR: available
Mar 21 10:47:47 jpenuchot-nzxt sunshine[1400]: [2022:03:21:10:47:47]: Info: -- Output --
Mar 21 10:47:47 jpenuchot-nzxt sunshine[1400]: [2022:03:21:10:47:47]: Info:   Resolution: 1920x1080
Mar 21 10:47:47 jpenuchot-nzxt sunshine[1400]: [2022:03:21:10:47:47]: Info:   Offset: 0x0
Mar 21 10:47:47 jpenuchot-nzxt sunshine[1400]: [2022:03:21:10:47:47]: Error: Couldn't destroy session handle: Unable to cleanup NvFBC
Mar 21 10:47:47 jpenuchot-nzxt sunshine[1400]: [2022:03:21:10:47:47]: Info: Screencasting with NvFBC
Mar 21 10:47:47 jpenuchot-nzxt sunshine[1400]: [2022:03:21:10:47:47]: Info: Color coding [Rec. 601]
Mar 21 10:47:47 jpenuchot-nzxt sunshine[1400]: [2022:03:21:10:47:47]: Info: Color range: [MPEG]
Mar 21 10:47:47 jpenuchot-nzxt sunshine[1400]: [2022:03:21:10:47:47]: Error: Failed to start capture session: Invalid external buffer capabilities
Mar 21 10:47:47 jpenuchot-nzxt sunshine[1400]: [2022:03:21:10:47:47]: Error: Couldn't destroy session handle: Unable to cleanup NvFBC
Mar 21 10:47:47 jpenuchot-nzxt sunshine[1400]: [2022:03:21:10:47:47]: Error: Couldn't release NvFBC context from current thread: Unable to cleanup NvFBC
```

Note: `nvidia-utils` was patched using `nvlax` through this package: https://aur.archlinux.org/packages/nvidia-utils-nvlax/

**Comment by KuleRucket** [comment, marked as answer]
Your problem is now this:
"Error: Failed to start capture session: Invalid external buffer capabilities"

see #100 

If you downgrade vulkan as in my comment, it should work.

**Reply by JPenuchot** [reply]
Thanks!

**Comment by IceDBorn** [comment]
I installed nvidia-utils-nvlax and downgraded vulkan-icd-loader and it does not fix the framerate issues.

**Reply by IceDBorn** [reply]
I did not get any errors with the up to date vulkan icd loader though. It seems like sunshine does not use it at all. A screen recorder that requires NvFBC crashed with the latest vulkan icd loader.

**Reply by IceDBorn** [reply]
@ReenigneArcher

**Reply by JPenuchot** [reply]
What version did you downgrade to? And btw you may try using `nvlax-git` instead of `nvidia-utils-nvlax`, it only contains `nvlax` and a pair of pacman hooks instead of a full nvidia-utils package. That's what I ended up using at last.

**Reply by IceDBorn** [reply]
vulkan-icd-loader-1.2.203-1

**Reply by JPenuchot** [reply]
And did you downgrade the other Vulkan packages such as vulkan headers, etc. as well?

**Reply by IceDBorn** [reply]
Is it really necessary? I was getting this exact same error "Error: Failed to start capture session: Invalid external buffer capabilities" with the screen recorder and it works fine by just downgrading vulkan-icd-loader

**Reply by IceDBorn** [reply]
I downgraded all the vulkan packages to the same version, the issue persists.

**Reply by JPenuchot** [reply]
Not sure it's necessary, but I'm not sure having mismatched Vulkan components is a good idea.

In my case I just ran `sudo downgrade lib32-vulkan-icd-loader vulkan-headers vulkan-icd-loader vulkan-tools` to downgrade the packages, rebooted, then rebuilt the `sunshine` AUR package.

Also I've added `export CUDA_PATH=/opt/cuda` at the end of my `/etc/makepkg.conf` to make sure CMake finds `nvcc`.

**Reply by IceDBorn** [reply]
Thanks that fixed the issue!
